plum/overload.py

```diff
@@ -0,0 +1,21 @@
+import sys
+from typing import Callable, TypeVar
+
+if sys.version_info >= (3, 11):  # pragma: specific no cover 3.7 3.8 3.9 3.10
+    from typing import get_overloads, overload
+else:  # pragma: specific no cover 3.11
+    from typing_extensions import get_overloads, overload
+
+from .function import Function
+
+__all__ = ["overload", "dispatch"]
+
+T = TypeVar("T", bound=Callable)
```

Comment:
With pyright (and by extension, pylance) in strict mode, `Callable` with no parameters will be flagged as an error.

<img width="573" alt="image" src="https://github.com/beartype/plum/assets/86978/83aaa4dc-9eb3-4fa6-a0c4-41a19f268a0f">

I believe that for mypy, `Callable` by itself is shorthand for `Callable[..., Any]`, so it should be like this:

```suggestion
T = TypeVar("T", bound=Callable[..., Any])
```

Reply:
Thanks for the suggestion, @wch! I've committed it. :)